contracts/uniswap_exchange.vy

```diff
@@ -248,13 +249,12 @@ def tokenToEthTransferExact(eth_bought: uint256(wei), max_tokens: uint256, deadl
 
 @private
 def tokenToToken(tokens_sold: uint256, min_tokens_bought: uint256, deadline: timestamp, buyer: address, recipient: address, exchange_addr: address) -> uint256:
-    assert self.totalSupply > 0 and deadline >= block.timestamp
-    assert exchange_addr != self
+    assert (self.totalSupply > 0 and deadline >= block.timestamp) and min_tokens_bought > 0
+    assert exchange_addr != self and exchange_addr != ZERO_ADDRESS
     eth_bought: uint256(wei) = self.tokenToEthPrice(tokens_sold)
     assert self.token.transferFrom(buyer, self, tokens_sold)
     # call fails if exchange_addr == ZERO_ADDRESS
     tokens_bought: uint256 = Exchange(exchange_addr).ethToTokenTransfer(min_tokens_bought, deadline, recipient, value=eth_bought)
-    assert tokens_bought > min_tokens_bought and min_tokens_bought > 0
```

Comment:
`tokens_bought > min_tokens_bought` is already checked inside `ethToTokenTransfer` function. whether or not we want to trust this external function call?